ansible_base/rbac/models.py

```diff
@@ -605,8 +605,10 @@ def accessible_ids(cls, model_cls, actor, codename: str, content_types: Optional
     @classmethod
     def accessible_objects(cls, model_cls, user, codename, queryset: Optional[QuerySet] = None) -> QuerySet:
         if queryset is None:
-            queryset = model_cls.objects
-        return queryset.filter(pk__in=cls.accessible_ids(model_cls, user, codename))
+            base_queryset = model_cls.objects
+        else:
+            base_queryset = queryset
+        return base_queryset.filter(pk__in=cls.accessible_ids(model_cls, user, codename))
```

Comment:
The else here feels "weird" to me. Out of curiosity, would the type checker be "happy" if instead of `model_cls.objects` we set `queryset = model_cls.objects.all()` (which would be a QuerySet). But I think that would be less efficient than just appending the filter to a Manager right?

Reply:
Interesting theory. I will give it a try. I'm not familiar enough with this to predict what would happen. No, I don't think that would be any less efficient. Every operation you make on a QuerySet produces a new queryset, but this is only python/in-memory considerations, which I'm committed to not caring about.

Reply:
implemented